sw/device/lib/dif/BUILD

```diff
@@ -0,0 +1,34 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+cc_library(
+  name = "clkmgr",
+  srcs = [
+      ":dif_clkmgr.c",
+      ":dif_clkmgr.h",
+      ":dif_warn_unused_result.h",
+      ],
+  deps = [
+      "//sw/device/lib/base:mmio",
+      "//hw/top_earlgrey/ip/clkmgr/data/autogen:clkmgr_regs",
+      ],
+  )
+
+cc_test(
+  name = "clkmgr_unittest",
+  srcs = [
+      ":dif_clkmgr_unittest.cc",
+      ],
+  deps = [
+      ":clkmgr",
+      "//hw/top_earlgrey/ip/clkmgr/data/autogen:clkmgr_regs",
```

Comment:
If the above dependency on the top_earlgrey clkmgr data is preserved, then this dependency is redundant and uneccesary.

Reply:
I'm not certain about that.  If `dif_clkmgr_unittest.cc` itself includes `clkmgr_regs.h`, I'm not certain it gets to #include the deps of its dependents.   Might also be a difference in blaze vs bazel.

I usually let `build_cleaner` help me with this sort of thing, but I don't think that tool is available outside of google3.

Reply:
It looks like ```build_cleaner``` is only supported for google3 and Blaze, without a plan to support Bazel or external projects. We could mirror lowRisc/OpenTitan in google3, but I'm not sure how far that would get us.

Reply:
We should not attempt to mirror into google3 - our project environment is too unique and trying to drop the code into g3 just to get `build_cleaner` to work for us is not worth the effort (bazel/blaze differences, toolchain setup, half the project isn't built by `bazel`, coding-style differences, etc).